setup.py

```diff
@@ -35,14 +35,93 @@ def is_conda_interpreter():
     """
     return 'conda' in sys.version or 'Continuum' in sys.version
 
+def is_macOS():
+    return 'darwin' in sys.platform
+
+
+# HOMEBREW_SAMPLE = """$ brew ls --verbose libomp
+# /opt/homebrew/Cellar/libomp/15.0.7/INSTALL_RECEIPT.json
+# /opt/homebrew/Cellar/libomp/15.0.7/.brew/libomp.rb
+# /opt/homebrew/Cellar/libomp/15.0.7/include/ompt.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp-tools.h
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.dylib
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.a
+# """
+#
+# MACPORTS_SAMPLE = """$ port contents libomp
+# Port libomp contains:
+#   /opt/local/include/libomp/omp-tools.h
+#   /opt/local/include/libomp/omp.h
+#   /opt/local/include/libomp/ompt.h
+#   /opt/local/lib/libomp/libgomp.dylib
+#   /opt/local/lib/libomp/libiomp5.dylib
+#   /opt/local/lib/libomp/libomp.dylib
+#   /opt/local/share/doc/libomp/LICENSE.TXT
+#   /opt/local/share/doc/libomp/README.txt
+# """
+
+def macOS_list_brew_port(cmd):
+    from subprocess import run
+    lib = run(cmd + " | awk '/\/lib\// { print $1; }' | xargs -n1 dirname | uniq",
+              shell=True, check=False, capture_output=True)
+    inc = run(cmd + " | awk '/\/include\// { print $1; }' | xargs -n1 dirname | uniq",
+              shell=True, check=False, capture_output=True)
+    if lib.returncode == 0 and inc.returncode == 0:
+        return inc.stdout.decode("UTF-8").strip(), lib.stdout.decode("UTF-8").strip()
+    return None, None
```

Comment:
You've unfortunately made a pull request for me to review the day after I attended a python meetup presentation on clean code...

There is repeated code here and it seems like the include and lib portions could be separated. Also, is it possible to keep this mostly in python-land? Maintenance is harder for those unfamiliar with bash tools like awk/xargs/uniq (although it isn't too hard to guess what's going on here). Plus if it is all in python you only have to run `cmd` once and manipulate the string after that.

Side question: Is `xargs` standard on OSX?

Reply:
Also thought about reducing this to an exercise in regex smashing, that's probably the more pythonic way to go. And yes, xargs and awk are system level, though you often have to stick to BSD without any GNU extensions when using those kinds of things. (At least we finally got `readlink -f` in Ventura!)

I'll make it collect output and change the awk-xargs-dirname-uniq to regex/pathlib/set mechanics, I can drop shell=True at that point and reduce the possibility of breakage due to environment pollution.